plugins/module_utils/network/sonic/config/mac/mac.py

```diff
@@ -0,0 +1,430 @@
+PATCH = 'patch'
+DELETE = 'delete'
+TEST_KEYS = [
+    {'config': {'vrf_name': ''}},
+    {'mac_table_entries': {'mac_address': '', 'vlan_id': ''}}
+]
+
+
+class Mac(ConfigBase):
+    """
+    The sonic_mac class
+    """
+
+    gather_subset = [
+        '!all',
+        '!min',
+    ]
+
+    gather_network_resources = [
+        'mac',
+    ]
+
+    def __init__(self, module):
+        super(Mac, self).__init__(module)
+
+    def get_mac_facts(self):
+        """ Get the 'facts' (the current configuration)
+
+        :rtype: A dictionary
+        :returns: The current configuration as a dictionary
+        """
+        facts, _warnings = Facts(self._module).get_facts(self.gather_subset, self.gather_network_resources)
+        mac_facts = facts['ansible_network_resources'].get('mac')
+        if not mac_facts:
+            return []
+        return mac_facts
+
+    def execute_module(self):
+        """ Execute the module
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        result = {'changed': False}
+        warnings = []
+        commands = []
+
+        existing_mac_facts = self.get_mac_facts()
+        commands, requests = self.set_config(existing_mac_facts)
+        if commands and len(requests) > 0:
+            if not self._module.check_mode:
+                try:
+                    edit_config(self._module, to_request(self._module, requests))
+                except ConnectionError as exc:
+                    self._module.fail_json(msg=str(exc), code=exc.code)
+            result['changed'] = True
+        result['commands'] = commands
+
+        changed_mac_facts = self.get_mac_facts()
+
+        result['before'] = existing_mac_facts
+        if result['changed']:
+            result['after'] = changed_mac_facts
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_mac_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        have = existing_mac_facts
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+        state = self._module.params['state']
+
+        diff = get_diff(want, have, TEST_KEYS)
```

Comment:
Please move get_diff to _state_merged as it is only one need diif.

Reply:
Addressed